### PR TITLE
Add cmake build configuration

### PR DESCRIPTION
**Pull request by p12tic**
cc @simogasp

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+# SOFTWARE.
+
+cmake_minimum_required(VERSION 3.11)
+
+project(Osi LANGUAGES C CXX)
+
+set(PROJECT_VERSION 0.107.9)
+set(PROJECT_VERSION_MAJOR 0)
+set(PROJECT_VERSION_MINOR 107)
+set(PROJECT_VERSION_PATCH 9)
```
Comment (simogasp) [inline CMakeLists.txt]:
```suggestion
project(Osi LANGUAGES C CXX VERSION 0.107.9)
```
like that CMake automatically generates the various `PROJECT_VERSION*` variables

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    "${CMAKE_CXX_FLAGS} -Wno-inconsistent-missing-override -Wno-unused-command-line-argument -Wno-unused-result -Wno-exceptions"
+    )
+  set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9"
+      CACHE STRING "Minimum OS X deployment version")
+endif()
+
+find_package(CoinUtils REQUIRED)
+find_package(Mosek)
+
+file(GLOB OSI_SOURCES src/Osi/*.hpp src/Osi/*.cpp)
```
Comment (simogasp) [inline CMakeLists.txt]: A bit of nitpicking but I'm not fan of the GLOB thing, it's always better to have something explicit like
```cmake
set(OSI_HEADERS
    "src/Osi/blabla.h"
    "...)

set(OSI_SOURCES
    "src/Osi/blabla.c"
    ...)

add_library(${PROJECT_NAME} ${OSI_SOURCES} ${OSI_HEADERS})

set_target_properties(${PROJECT_NAME}  PROPERTIES
    PUBLIC_HEADER ${OSI_HEADERS} ...)
```

**Review by simogasp — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (p12tic) [reply CMakeLists.txt]: Fixed now.

File: CMakeLists.txt (hunk 2 above)
Comment (p12tic) [reply CMakeLists.txt]: Fixed now.

**Comment by p12tic** [comment]
@simogasp I've addressed your comments

**Review by simogasp — APPROVED** [review]
great!

**Pull request merged**